providers/azure/auth.go

```diff
@@ -0,0 +1,57 @@
+package azure
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+const (
+	credentialsPath = "/.azure/credentials.json"
+)
+
+// ToJSON returns the passed item as a pretty-printed JSON string. If any JSON error occurs,
+// it returns the empty string.
+func ToJSON(v interface{}) (string, error) {
+	j, err := json.MarshalIndent(v, "", "  ")
+	return string(j), err
+}
```

Comment:
it might be easier to create a `MarshalJSON` method on the types that you want to pretty-print, so that they conform to [`Marshaler`](https://godoc.org/encoding/json#Marshaler). Then you can `json.Marshal` them as usual and they'll always be pretty-printed.

just an idea, not blocking